omegaconf/_utils.py

```diff
@@ -217,13 +220,16 @@ def get_attr_data(obj: Any, allow_objects: Optional[bool] = None) -> Dict[str, A
             )
             format_and_raise(node=None, key=None, value=value, cause=e, msg=str(e))
 
-        d[name] = _maybe_wrap(
-            ref_type=type_,
-            is_optional=is_optional,
-            key=name,
-            value=value,
-            parent=dummy_parent,
-        )
+        try:
+            d[name] = _maybe_wrap(
+                ref_type=type_,
+                is_optional=is_optional,
+                key=name,
+                value=value,
+                parent=dummy_parent,
+            )
+        except ValidationError as ex:
+            dummy_parent._format_and_raise(key=name, value=value, cause=ex)
```

Comment:
can you try to use `format_and_raise()` directly instead of through the dummy parent?
it can probably make things a bit cleaner when testing this.

Reply:
In https://github.com/omry/omegaconf/pull/559/commits/f2a1bfd2af08212cb2b7e294e24f63bdf8fb28a4 I've switched from `dummy_parent._format_and_raise(...)` to `format_and_raise(node=None, ...)`.

The disadvantage of this change is that we no longer have `object_type` information in the error message (as `dummy_parent` was providing the `object_type`).

Reply:
Besides the `object_type` issue, the error messages are also not reporting `full_key` accurately. I think that reporting on `full_key` would be easiest if we pass a `parent_node` into the `get_structured_config_data` function (so that `full_key` could be computed using the parent config). This change would require some plumbing of the methods that call `get_structured_config_data`. I can give it a try if you're interested...

Reply:
I think this will trivially easier once we have the structured_ir diff baked and ready.
for now this is good enough.